### PR TITLE
Perform some todo tasks for 0.18

### PR DESCRIPTION
**Pull request by sciunto**
## Description

I did some of the tasks for 0.18. The PR is already long to review. I stop here in the list.

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/data/__init__.py [hunk 1]
```diff
@@ -235,7 +232,6 @@ def _init_pooch():
     shutil.copy2(osp.join(skimage_distribution_dir, 'data', 'README.txt'),
                  osp.join(data_dir, 'README.txt'))
 
-    data_base_dir = osp.join(data_dir, '..')
```
Comment (soupault) [inline skimage/data/__init__.py:238]: Why this change?

File: skimage/segmentation/active_contour_model.py [hunk 2]
```diff
@@ -7,11 +6,11 @@
 
 def active_contour(image, snake, alpha=0.01, beta=0.1,
                    w_line=0, w_edge=1, gamma=0.01,
-                   bc=None, max_px_move=1.0,
+                   max_px_move=1.0,
                    max_iterations=2500, convergence=0.1,
                    *,
                    boundary_condition='periodic',
-                   coordinates=None):
+                   coordinates='rc'):
```
Comment (soupault) [inline skimage/segmentation/active_contour_model.py:13]: If I understood correctly the original motivation, the idea was to remove this argument once the deprecation is over - https://github.com/scikit-image/scikit-image/pull/4035 .

File: skimage/data/__init__.py (hunk 1 above)
Comment (sciunto) [reply skimage/data/__init__.py:238]: no idea :)

File: skimage/segmentation/active_contour_model.py (hunk 2 above)
Comment (sciunto) [reply skimage/segmentation/active_contour_model.py:13]: Thank you!

Comment (sciunto) [reply skimage/segmentation/active_contour_model.py:13]: My concern is: if I remove the parameter, it will break the user code because the message said to put the option ='rc' to suppress the warning.
Ideally, we need another deprecation round to raise a warning and say, to suppress it, remove this kw arg.

@jni any comment on your original thought?

Comment (emmanuelle) [reply skimage/segmentation/active_contour_model.py:13]: So we would need to add another deprecation message here?

Comment (emmanuelle) [reply skimage/segmentation/active_contour_model.py:13]: the gallery example `edges/plot_active_contour.py` uses `coordinates='rc'` and should be modified too.

Comment (sciunto) [reply skimage/segmentation/active_contour_model.py:13]: yes

File: skimage/segmentation/active_contour_model.py [hunk 3]
```diff
@@ -97,22 +94,6 @@ def active_contour(image, snake, alpha=0.01, beta=0.1,
...
-                   'from xy coordinates (transposed from image dimensions) to '
-                   'rc coordinates in scikit-image 0.18. Set '
-                   "`coordinates='rc'` to silence this warning. "
-                   "`coordinates='xy'` will restore the old behavior until "
-                   '0.18, but will stop working thereafter.')
-        warn(message, category=FutureWarning, stacklevel=2)
-        coordinates = 'xy'
-        snake_xy = snake
     if coordinates == 'rc':
         snake_xy = snake[:, ::-1]
```
Comment (jni) [inline skimage/segmentation/active_contour_model.py:98]: This if statement needs to be removed. My suggestion is that we leave `coordinates` as a do-nothing argument until version 1.0, at which point we can remove it. Potentially, we should raise a ValueError if coordinates is anything *other* than 'rc' now.

This kind of 4-version deprecation would be best to avoid — ie we don't want to tell people, use this argument to avoid warnings. Ok now stop using this argument altogether. So, I'm happy to leave coordinates around, but all values other than 'rc' should be an error.

File: skimage/segmentation/active_contour_model.py (hunk 2 above)
Comment (jni) [reply skimage/segmentation/active_contour_model.py:13]: @emmanuelle @soupault as per my comment below, I think this argument should stay but it should do nothing, other than maybe raise a ValueError when set to anything other than 'rc'. Indeed, it *must* raise a ValueError to remind anyone using `coordinates='xy'` that their code is no longer valid.

File: skimage/segmentation/active_contour_model.py (hunk 3 above)
Comment (sciunto) [reply skimage/segmentation/active_contour_model.py:98]: Thanks for the clarification. I'll do that.

File: skimage/segmentation/tests/test_active_contour_model.py [hunk 4]
```diff
@@ -118,29 +117,3 @@ def test_bad_input():
...
-        c = 220 + 100*np.cos(s)
-        init = np.array([r, c]).T
-        snake = active_contour(gaussian(img, 3), init,
-                               bc='periodic', alpha=0.015, beta=10,
-                               w_line=0, w_edge=1, gamma=0.001,
-                               max_iterations=100, coordinates='rc')
-
-
-def test_xy_coord_warning():
-    # this should raise ValueError after 0.18.
```
Comment (jni) [inline skimage/segmentation/tests/test_active_contour_model.py:136]: @sciunto as per this comment, this test should be brought back with `with testing.raises(ValueError)`

**Comment by sciunto** [comment]
Please, comment on this last changes. I put a note in TODO.txt to remove the option for 1.0.

File: skimage/segmentation/tests/test_active_contour_model.py [hunk 5]
```diff
@@ -112,35 +104,27 @@ def test_bad_input():
...
 
 
-def test_bc_deprecation():
-    with expected_warnings(['boundary_condition']):
-        img = rgb2gray(data.astronaut())
-        s = np.linspace(0, 2*np.pi, 400)
-        r = 100 + 100*np.sin(s)
-        c = 220 + 100*np.cos(s)
-        init = np.array([r, c]).T
+def test_xy_coord_warning():
```
Comment (soupault) [inline skimage/segmentation/tests/test_active_contour_model.py:112]: I would use `for` loop to test for `xy` and `None` and rename the test:
```suggestion
def test_coord_raises():
```

File: skimage/segmentation/active_contour_model.py [hunk 6]
```diff
@@ -54,9 +54,9 @@ def active_contour(image, snake, alpha=0.01, beta=0.1,
         be combined by parsing 'fixed-free', 'free-fixed'. Parsing
         'fixed-fixed' or 'free-free' yields same behaviour as 'fixed' and
         'free', respectively.
-    coordinates : {'rc' or 'xy'}, optional
-        Whether to use rc or xy coordinates. The 'xy' option (current default)
-        will be removed in version 0.18.
+    coordinates : {'rc'}, optional
+        This option remains for compatibility purpose only and has no effect.
```
Comment (hmaarrfk) [inline skimage/segmentation/active_contour_model.py:58]:
```suggestion
        This option remains for compatibility purpose only and has no effect.
        It was introduced in 0.16 with the ``'xy'`` option, but since 0.18, 
        only the ``'rc'`` option is valid.
```

Comment (hmaarrfk) [reply skimage/segmentation/active_contour_model.py:58]: A little bit of history goes a long way to future us!

File: skimage/data/__init__.py [hunk 7]
```diff
@@ -316,31 +313,6 @@ def lbp_frontal_face_cascade_filename():
     return _fetch('data/lbpcascade_frontalface_opencv.xml')
 
 
-def load(f, as_gray=False):
```
Comment (hmaarrfk) [inline skimage/data/__init__.py:319]: Is it easy to add this back? I'm wondering if this might make it hard to backport any pooch fixes. But maybe we release 0.18 quickly?

Comment (hmaarrfk) [reply skimage/data/__init__.py:319]: Nevermind, this is the wrapper to the internal `_load` function. should be fine.

File: skimage/segmentation/active_contour_model.py [hunk 8]
```diff
@@ -97,24 +94,10 @@ def active_contour(image, snake, alpha=0.01, beta=0.1,
...
-                   '0.18, but will stop working thereafter.')
-        warn(message, category=FutureWarning, stacklevel=2)
-        coordinates = 'xy'
-        snake_xy = snake
-    if coordinates == 'rc':
-        snake_xy = snake[:, ::-1]
+    if coordinates != 'rc':
+        raise ValueError('Coordinate values must be set in a row column '
+                         'format. `coordinates` must be indifferently set '
+                         'to "rc" or None.')
```
Comment (jni) [inline skimage/segmentation/active_contour_model.py:100]: Well, this message is now incorrect: `coordinates=None` will raise this error!

Comment (sciunto) [reply skimage/segmentation/active_contour_model.py:102]: Sorry, I changed, rechanged, re-rechanged and forgot to re*3 change.

File: skimage/segmentation/active_contour_model.py [hunk 9]
```diff
@@ -97,24 +96,10 @@ def active_contour(image, snake, alpha=0.01, beta=0.1,
...
```
Comment (sciunto) [inline skimage/segmentation/active_contour_model.py:102]:
```suggestion
                         'to "rc".')
```

File: skimage/segmentation/active_contour_model.py [hunk 10]
```diff
@@ -97,24 +96,10 @@ def active_contour(image, snake, alpha=0.01, beta=0.1,
...
-                   "`coordinates='xy'` will restore the old behavior until "
-                   '0.18, but will stop working thereafter.')
-        warn(message, category=FutureWarning, stacklevel=2)
-        coordinates = 'xy'
-        snake_xy = snake
-    if coordinates == 'rc':
-        snake_xy = snake[:, ::-1]
+    if coordinates != 'rc':
+        raise ValueError('Coordinate values must be set in a row column '
+                         'format. `coordinates` must be indifferently set '
```
Comment (sciunto) [inline skimage/segmentation/active_contour_model.py:101]:
```suggestion
                         'format. `coordinates` must be set '
```

**Comment by sciunto** [comment]
@jni Is this PR OK for you?
